### DISCUSSION TITLE
CORS error ketika di production

### DISCUSSION THREAD

**Discussion by akbarism** [kirimwa-id/forum#54] [Tanya Jawab] [answered]
Saya mencoba API ini menggunakan VUE js di sisi client, dan Axios untuk http requestnya, selama development ini bekerja dengan sangat baik, tapi Saya tidak bisa menggunakan API ini di production, setelah web saya deploy pada netlify, saya mendapati CORS error pada palikasi saya.

![image](https://user-images.githubusercontent.com/59020048/136891591-48ff76f1-5533-4bbf-ae55-6d259df463ff.png)

**Comment by rioastamal** [comment, marked as answer]
API KirimWA.id sengaja tidak mengirim CORS header karena memang setiap API request memerlukan API Token. Jika API request menggunakan client side Javascript berarti API Token tersebut ter-ekspose disisi client  (browser) jadi pengguna aplikasi bisa "mencurinya" lewat Developer Tools atau View Source.

Apakah aplikasi ini nantinya untuk end-user lain atau digunakan untuk pribadi/sendiri saja?

**Reply by akbarism** [reply]
oke mas rio, terimakasih atas penjelasanya,
memang aplikasi yang saya buat sementara hanya untuk membantu beberapa teman yang perlu broadcast saja, sudah di handle menggunakan plugin, jadi overall ga ada masalah.
